aries_cloudagent/protocols/coordinate_mediation/v1_0/manager.py

```diff
@@ -74,11 +75,37 @@ async def _create_routing_did(self) -> DIDInfo:
         await storage.add_record(record)
         return info
 
-    async def receive_request(self, request: MediationRequest) -> MediationRecord:
-        """Create a new mediation record to track this request."""
+    async def receive_request(self,
+                              conn_id: str,
+                              request: MediationRequest
+                              ) -> MediationRecord:
+        """Create a new mediation record to track external request."""
+        role = MediationRecord.ROLE_SERVER
+        record = await self._store_request(request, conn_id, role)
+        return record
+
+    async def prepare_request(self,
+                              conn_id: str,
+                              request: MediationRequest
+                              ) -> MediationRecord:
+        """Create a new mediation record to track internal request."""
+        role = MediationRecord.ROLE_CLIENT
+        record = await self._store_request(request, conn_id, role)
+        return record
```

Comment:
Let's leave this stubbed out for now.

Reply:
I need this for routes.py. Should I move the logic into the admin handler?

Reply:
As per our offline discussion, I have some internal conflict on this point but for now I'll concede. Carry on!